examples/autoignition-mpi.py

```diff
@@ -0,0 +1,229 @@
+    can_p = cantera_soln.P
+
+    casename = "autoignition"
+    pyrometheus_mechanism = pyro.get_thermochem_class(cantera_soln)(actx.np)
+    eos = PyrometheusMixture(pyrometheus_mechanism, tguess=init_temperature)
+
+    print(f"Cantera state (rho,T,P,Y) = ({can_rho}, {can_t}, {can_p}, {can_y}")
+    initializer = MixtureInitializer(dim=dim, nspecies=nspecies,
+                                     pressure=can_p, temperature=can_t,
+                                     massfractions=can_y, velocity=velocity)
+
+    my_boundary = AdiabaticSlipBoundary()
+    boundaries = {BTAG_ALL: my_boundary}
+    current_state = initializer(eos=eos, x_vec=nodes, t=0)
+
+    # Inspection at physics debugging time
+    #    cv = split_conserved(dim, current_state)
+    #    print(f"Initial CV rho: {cv.mass}")
+    #    print(f"Initial CV rhoE: {cv.energy}")
+    #    print(f"Initial CV rhoV: {cv.momentum}")
+    #    print(f"Initial CV rhoY: {cv.mass_fractions}")
+    #    print(f"Initial Y: {cv.mass_fractions / cv.mass}")
+    #    print(f"Initial DV pressure: {eos.pressure(cv)}")
+    #    print(f"Initial DV temperature: {eos.temperature(cv)}")
+
+    visualizer = make_visualizer(discr, discr.order + 3
+                                 if discr.dim == 2 else discr.order)
+    initname = initializer.__class__.__name__
+    eosname = eos.__class__.__name__
+    init_message = make_init_message(dim=dim, order=order,
+                                     nelements=local_nelements,
+                                     global_nelements=global_nelements,
+                                     dt=current_dt, t_final=t_final, nstatus=nstatus,
+                                     nviz=nviz, cfl=current_cfl,
+                                     constant_cfl=constant_cfl, initname=initname,
+                                     eosname=eosname, casename=casename)
+    cantera_soln.equilibrate("UV")
+    can_eq_t, can_eq_rho, can_eq_y = cantera_soln.TDY
+    can_eq_p = cantera_soln.P
+    #    can_eq_e = cantera_soln.int_energy_mass
+    #    can_eq_k = cantera_soln.forward_rate_constants
+    #    can_eq_c = cantera_soln.concentrations
+    #    can_eq_r = cantera_soln.net_rates_of_progress
+    #    can_eq_omega = cantera_soln.net_production_rates
+    if rank == 0:
+        logger.info(init_message)
+        logger.info(f"Expected (p,T,rho, y) = ({can_eq_p}, {can_eq_t},"
+                    f" {can_eq_rho}, {can_eq_y})")
+
+    get_timestep = partial(inviscid_sim_timestep, discr=discr, t=current_t,
+                           dt=current_dt, cfl=current_cfl, eos=eos,
+                           t_final=t_final, constant_cfl=constant_cfl)
+
+    def my_rhs(t, state):
+        cv = split_conserved(dim=dim, q=state)
+        return (inviscid_operator(discr, q=state, t=t,
+                                  boundaries=boundaries, eos=eos)
+                + eos.get_species_source_terms(cv))
+
+    def my_checkpoint(step, t, dt, state):
+        global checkpoint_t
```

Comment:
I don't think this works as you intend. Read up on `nonlocal` or boxed values. Maybe discuss at next meeting.

Reply:
No, it certainly doesn't.

Reply:
Yeah, we can talk about it.  I have fixed this, but I think the intent is misunderstood in-general.  The issue is that if the simulation stops on a step that was not an output step, then we need to output in order to capture the final state of the sim.  Any one of several situations can cause stepping to exit on a non-output step.  Writing the final state is necessary for restarting from where you stopped, and/or debugging the flow.

Reply:
Fixed in (f2bae70)